syntax = "proto3";

package yandex.cloud.audit.compute.instancegroup;

import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";
import "google/rpc/status.proto";
import "yandex/cloud/audit/common.proto";
import "yandex/cloud/validation.proto";
import "yandex/cloud/compute/v1/instancegroup/instance_group.proto";

option go_package = "a.yandex-team.ru/cloud/bitbucket/public-api/yandex/cloud/audit/compute/instancegroup;instancegroup";
option java_package = "yandex.cloud.api.audit.compute.instancegroup";
option java_outer_classname = "InstanceGroupEvents";


message CreateInstanceGroup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2;
    yandex.cloud.compute.v1.instancegroup.InstanceTemplate instance_template = 3;
    yandex.cloud.compute.v1.instancegroup.ScalePolicy scale_policy = 4;
    yandex.cloud.compute.v1.instancegroup.DeployPolicy deploy_policy = 5;
    yandex.cloud.compute.v1.instancegroup.AllocationPolicy allocation_policy = 6;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerSpec load_balancer_spec = 7;
    yandex.cloud.compute.v1.instancegroup.HealthChecksSpec health_checks_spec = 8;
    string service_account_id = 9;
    string status = 10;
    repeated yandex.cloud.compute.v1.instancegroup.Variable variables = 11;
    bool deletion_protection = 12;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerSpec application_load_balancer_spec = 13;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerState load_balancer_state = 14;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerState application_load_balancer_state = 15;
  }
}

message UpdateInstanceGroup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2;
    yandex.cloud.compute.v1.instancegroup.InstanceTemplate instance_template = 3;
    yandex.cloud.compute.v1.instancegroup.ScalePolicy scale_policy = 4;
    yandex.cloud.compute.v1.instancegroup.DeployPolicy deploy_policy = 5;
    yandex.cloud.compute.v1.instancegroup.AllocationPolicy allocation_policy = 6;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerSpec load_balancer_spec = 7;
    yandex.cloud.compute.v1.instancegroup.HealthChecksSpec health_checks_spec = 8;
    string service_account_id = 9;
    string status = 10;
    repeated yandex.cloud.compute.v1.instancegroup.Variable variables = 11;
    bool deletion_protection = 12;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerSpec application_load_balancer_spec = 13;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerState load_balancer_state = 14;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerState application_load_balancer_state = 15;
  }
}

message StartInstanceGroup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
    yandex.cloud.compute.v1.instancegroup.InstanceTemplate instance_template = 3;
    yandex.cloud.compute.v1.instancegroup.ScalePolicy scale_policy = 4;
    yandex.cloud.compute.v1.instancegroup.DeployPolicy deploy_policy = 5;
    yandex.cloud.compute.v1.instancegroup.AllocationPolicy allocation_policy = 6;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerSpec load_balancer_spec = 7;
    yandex.cloud.compute.v1.instancegroup.HealthChecksSpec health_checks_spec = 8;
    string service_account_id = 9;
    string status = 10;
    repeated yandex.cloud.compute.v1.instancegroup.Variable variables = 11;
    bool deletion_protection = 12;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerSpec application_load_balancer_spec = 13;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerState load_balancer_state = 14;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerState application_load_balancer_state = 15;
  }
}

message StopInstanceGroup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
    yandex.cloud.compute.v1.instancegroup.InstanceTemplate instance_template = 3;
    yandex.cloud.compute.v1.instancegroup.ScalePolicy scale_policy = 4;
    yandex.cloud.compute.v1.instancegroup.DeployPolicy deploy_policy = 5;
    yandex.cloud.compute.v1.instancegroup.AllocationPolicy allocation_policy = 6;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerSpec load_balancer_spec = 7;
    yandex.cloud.compute.v1.instancegroup.HealthChecksSpec health_checks_spec = 8;
    string service_account_id = 9;
    string status = 10;
    repeated yandex.cloud.compute.v1.instancegroup.Variable variables = 11;
    bool deletion_protection = 12;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerSpec application_load_balancer_spec = 13;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerState load_balancer_state = 14;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerState application_load_balancer_state = 15;
  }
}

message DeleteInstanceGroup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
    yandex.cloud.compute.v1.instancegroup.InstanceTemplate instance_template = 3;
    yandex.cloud.compute.v1.instancegroup.ScalePolicy scale_policy = 4;
    yandex.cloud.compute.v1.instancegroup.DeployPolicy deploy_policy = 5;
    yandex.cloud.compute.v1.instancegroup.AllocationPolicy allocation_policy = 6;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerSpec load_balancer_spec = 7;
    yandex.cloud.compute.v1.instancegroup.HealthChecksSpec health_checks_spec = 8;
    string service_account_id = 9;
    string status = 10;
    repeated yandex.cloud.compute.v1.instancegroup.Variable variables = 11;
    bool deletion_protection = 12;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerSpec application_load_balancer_spec = 13;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerState load_balancer_state = 14;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerState application_load_balancer_state = 15;
  }
}

message SetInstanceGroupAccessBindings {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
    repeated AccessBinding access_bindings = 3;
    repeated AccessBindingDelta access_binding_deltas = 4;
  }
}

message UpdateInstanceGroupAccessBindings {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
    repeated AccessBinding access_bindings = 3;
    repeated AccessBindingDelta access_binding_deltas = 4;
  }
}

message PauseInstanceGroup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
    yandex.cloud.compute.v1.instancegroup.InstanceTemplate instance_template = 3;
    yandex.cloud.compute.v1.instancegroup.ScalePolicy scale_policy = 4;
    yandex.cloud.compute.v1.instancegroup.DeployPolicy deploy_policy = 5;
    yandex.cloud.compute.v1.instancegroup.AllocationPolicy allocation_policy = 6;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerSpec load_balancer_spec = 7;
    yandex.cloud.compute.v1.instancegroup.HealthChecksSpec health_checks_spec = 8;
    string service_account_id = 9;
    string status = 10;
    repeated yandex.cloud.compute.v1.instancegroup.Variable variables = 11;
    bool deletion_protection = 12;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerSpec application_load_balancer_spec = 13;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerState load_balancer_state = 14;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerState application_load_balancer_state = 15;
  }
}

message ResumeInstanceGroup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
    yandex.cloud.compute.v1.instancegroup.InstanceTemplate instance_template = 3;
    yandex.cloud.compute.v1.instancegroup.ScalePolicy scale_policy = 4;
    yandex.cloud.compute.v1.instancegroup.DeployPolicy deploy_policy = 5;
    yandex.cloud.compute.v1.instancegroup.AllocationPolicy allocation_policy = 6;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerSpec load_balancer_spec = 7;
    yandex.cloud.compute.v1.instancegroup.HealthChecksSpec health_checks_spec = 8;
    string service_account_id = 9;
    string status = 10;
    repeated yandex.cloud.compute.v1.instancegroup.Variable variables = 11;
    bool deletion_protection = 12;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerSpec application_load_balancer_spec = 13;
    yandex.cloud.compute.v1.instancegroup.LoadBalancerState load_balancer_state = 14;
    yandex.cloud.compute.v1.instancegroup.ApplicationLoadBalancerState application_load_balancer_state = 15;
  }
}

message StopInstanceGroupInstances {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
  }
}

message DeleteInstanceGroupInstances {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  enum EventStatus {
    EVENT_STATUS_UNSPECIFIED = 0;
    STARTED = 1;
    ERROR = 2;
    DONE = 3;
  }

  message EventDetails {
    string instance_group_id = 1 [(required) = true];
    string instance_group_name = 2 [(required) = true];
  }
}
